src/components/Sidebar/index.tsx

```diff
@@ -0,0 +1,175 @@
+import { useState, type ReactNode } from 'react';
+
+import { AppRegistration, AssignmentAdd, Checklist, Home, PeopleAlt } from '@mui/icons-material';
+import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
+import MenuIcon from '@mui/icons-material/Menu';
+import { Fade, useMediaQuery } from '@mui/material';
+import Divider from '@mui/material/Divider';
+import Drawer from '@mui/material/Drawer';
+import IconButton from '@mui/material/IconButton';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+
+// Wrapper
+const DrawerWrapper = ({ children, isOpen, isMobile }: { children: ReactNode[], isOpen: boolean, isMobile: boolean }) => {
+  const drawerWidth = isMobile ? '100%' : '300px';
+
+  return (
+    <Drawer
+      sx={{
+        width: isOpen ? drawerWidth : '80px',
+        flexShrink: 0,
+        overflow: 'hidden',
+        transition: '.5s all ease',
+        '& .MuiDrawer-paper': {
+          width: isOpen ? drawerWidth : '80px',
+          padding: '1em',
+          boxSizing: 'border-box',
+          border: 'none',
+          boxShadow: '1px 0px 5px #efefef',
+          transition: '.5s all ease',
+        },
+      }}
+      variant="persistent"
+      anchor="left"
+      open={isMobile ? isOpen : true}
```

Comment:
Maybe for smaller computers it would also be nice to have it 'toggleable'
```suggestion
      open={isOpen}
```